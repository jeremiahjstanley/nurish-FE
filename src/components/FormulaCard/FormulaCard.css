div .formula-card {
  width: 100%;
  height: 100%;
  border-radius: 30px;
  padding: 5px 15px; }

.formula-link {
  width: 23%;
  height: 350px;
  margin: 5px; }

h2 {
  font-size: 1.4rem; }

.box .line-clamp-box {
  font-size: 1rem;
  text-align: left;
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical !important;
  overflow: hidden; }

#paragraph {
  -webkit-box-orient: vertical;
  display: -webkit-box;
  -webkit-line-clamp: 5;
  overflow: hidden; }

ul {
  padding: 0;
  list-style-type: none;
  font-size: 1rem;
  text-align: left; }

li {
  font-size: 1rem;
  text-align: left; }
