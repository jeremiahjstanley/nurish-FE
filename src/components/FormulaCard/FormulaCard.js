import React from 'react';
import PropTypes from 'prop-types';


export const FormulaCard = (formula) => {

  const usageGuidelines = () => {
    return formula.usage.map(usage => {
      return (
          <h5>{usage}</h5>
        )
    })
  }
  return (
    <div className='formula-card' key={formula.id}>
      <h2>{formula.title}</h2>
      <h4>{formula.description}</h4>
      <h4>usage: {usageGuidelines()}</h4>
    </div>
    )
}

FormulaCard.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string,
  description: PropTypes.string,
  usage: PropTypes.array
}