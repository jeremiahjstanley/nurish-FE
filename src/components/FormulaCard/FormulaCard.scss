div .formula-card {
  width: 100%;
  height: 100%;
  border-radius: 30px;
  padding: 5px 15px;
}

.formula-link {
  width: 23%;
  height: 350px;
  margin: 5px;
}

h2 {
  font-size: 1.4rem;
}

.box .line-clamp-box { 
  font-size: 1rem;
  text-align: left;
  // height: 100px
}

#paragraph {              
  // display: -webkit-box;                                
  -webkit-box-orient: vertical;  
  display: -webkit-box;
  -webkit-line-clamp: 5;
  overflow: hidden;
  // line-height: 1.45
}


ul {
  padding: 0;
  list-style-type: none;
  font-size: 1rem;
  text-align: left;
}

li {
  font-size: 1rem;
  text-align: left;
  // display: -webkit-box;
  // -webkit-line-clamp: 3;
  // overflow: hidden;
  // -webkit-box-orient: vertical; 
}
