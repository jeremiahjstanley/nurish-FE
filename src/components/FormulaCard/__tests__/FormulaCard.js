import React from 'react';
import { shallow } from 'enzyme';
import { FormulaCard } from '../FormulaCard.js';

describe('FormulaCard', () => {
  let wrapper
  let mockUsage

  beforeEach(() => {

  	mockUsage = 'Breakfast Substition, Renal Failure'

    wrapper = shallow(<FormulaCard 
    		usage={mockUsage}
    	/>)
  });

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

});