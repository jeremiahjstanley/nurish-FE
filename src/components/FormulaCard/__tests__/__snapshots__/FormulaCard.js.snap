// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormulaCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    usage="Breakfast Substition, Renal Failure"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="formula-card"
>
        <h2 />
        <div
                alt=""
                className="formula-card-image"
                style={
                        Object {
                                "backgroundImage": "url(/images/undefined)",
                              }
                }
        />
        <div
                className="formula-description"
        >
                <p
                        className="truncate-description"
                />
        </div>
        <p
                className="usage-header"
        >
                <strong>
                        usage:
                </strong>
        </p>
        <ul>
                <li>
                        Breakfast Substition
                </li>
                <li>
                         Renal Failure
                </li>
        </ul>
</div>,
      "className": "formula-link",
      "replace": false,
      "to": "/formulas/undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": "formula-card-undefined",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2 />,
          <div
            alt=""
            className="formula-card-image"
            style={
                        Object {
                                    "backgroundImage": "url(/images/undefined)",
                                  }
            }
/>,
          <div
            className="formula-description"
>
            <p
                        className="truncate-description"
            />
</div>,
          <p
            className="usage-header"
>
            <strong>
                        usage:
            </strong>
</p>,
          <ul>
            <li>
                        Breakfast Substition
            </li>
            <li>
                         Renal Failure
            </li>
</ul>,
        ],
        "className": "formula-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "formula-card-image",
            "style": Object {
              "backgroundImage": "url(/images/undefined)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="truncate-description"
/>,
            "className": "formula-description",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "truncate-description",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              usage:
</strong>,
            "className": "usage-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "usage:",
            },
            "ref": null,
            "rendered": "usage:",
            "type": "strong",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                Breakfast Substition
</li>,
              <li>
                 Renal Failure
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "formula-card-undefined-/Breakfast Substition",
              "nodeType": "host",
              "props": Object {
                "children": "Breakfast Substition",
              },
              "ref": null,
              "rendered": "Breakfast Substition",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "formula-card-undefined-/ Renal Failure",
              "nodeType": "host",
              "props": Object {
                "children": " Renal Failure",
              },
              "ref": null,
              "rendered": " Renal Failure",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="formula-card"
>
          <h2 />
          <div
                    alt=""
                    className="formula-card-image"
                    style={
                              Object {
                                        "backgroundImage": "url(/images/undefined)",
                                      }
                    }
          />
          <div
                    className="formula-description"
          >
                    <p
                              className="truncate-description"
                    />
          </div>
          <p
                    className="usage-header"
          >
                    <strong>
                              usage:
                    </strong>
          </p>
          <ul>
                    <li>
                              Breakfast Substition
                    </li>
                    <li>
                               Renal Failure
                    </li>
          </ul>
</div>,
        "className": "formula-link",
        "replace": false,
        "to": "/formulas/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": "formula-card-undefined",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <div
              alt=""
              className="formula-card-image"
              style={
                            Object {
                                          "backgroundImage": "url(/images/undefined)",
                                        }
              }
/>,
            <div
              className="formula-description"
>
              <p
                            className="truncate-description"
              />
</div>,
            <p
              className="usage-header"
>
              <strong>
                            usage:
              </strong>
</p>,
            <ul>
              <li>
                            Breakfast Substition
              </li>
              <li>
                             Renal Failure
              </li>
</ul>,
          ],
          "className": "formula-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "",
              "className": "formula-card-image",
              "style": Object {
                "backgroundImage": "url(/images/undefined)",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p
                className="truncate-description"
/>,
              "className": "formula-description",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "truncate-description",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                usage:
</strong>,
              "className": "usage-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "usage:",
              },
              "ref": null,
              "rendered": "usage:",
              "type": "strong",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                  Breakfast Substition
</li>,
                <li>
                   Renal Failure
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "formula-card-undefined-/Breakfast Substition",
                "nodeType": "host",
                "props": Object {
                  "children": "Breakfast Substition",
                },
                "ref": null,
                "rendered": "Breakfast Substition",
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "formula-card-undefined-/ Renal Failure",
                "nodeType": "host",
                "props": Object {
                  "children": " Renal Failure",
                },
                "ref": null,
                "rendered": " Renal Failure",
                "type": "li",
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
